Sources/Prometheus/PrometheusMetrics.swift

```diff
@@ -64,6 +64,37 @@ private class MetricsHistogram: RecorderHandler {
     }
 }
 
+/// This is a `swift-metrics` timer backed by a Prometheus' `PromHistogram` implementation.
+/// This is superior to `Summary` backed timer as `Summary` emits a set of quantiles, which will be impossible to correctly aggregate when one wants to render a percentile for a set of multiple instances.
+/// `Histogram` aggregation is possible with some fancy server-side math.
+class MetricsHistogramTimer: TimerHandler {
+    let histogram: PromHistogram<Int64, DimensionHistogramLabels>
+    let labels: DimensionHistogramLabels?
+    // this class is a lightweight wrapper around heavy prometheus metric type. This class is not cached and each time
+    // created anew. This allows us to use variable timeUnit without locking.
+    var timeUnit: TimeUnit?
+
+    init(histogram: PromHistogram<Int64, DimensionHistogramLabels>, dimensions: [(String, String)]) {
+        self.histogram = histogram
+        if !dimensions.isEmpty {
+            self.labels = DimensionHistogramLabels(dimensions)
+        } else {
+            self.labels = nil
+        }
+    }
+
+    // this is questionable as display unit here affects how the data is stored, and not how it's observed.
+    // should we delete it and tell preferDisplayUnit is not supported?
+    func preferDisplayUnit(_ unit: TimeUnit) {
+        self.timeUnit = unit
+    }
```

Comment:
Instead of somewhat supporting `displayUnit`, instead `Histogram` should be updated to take a `displayUnit` into account similar to `Summary`.

Reply:
This would be much more difficult as `Summary` produces aggregates on demand, while `Histogram` precomputes values when metric is reported. So changing timeUnit in `Summary` will allow you to produce different time units on the fly, while changing it in `Histogram` will force you to modify data during the reporting. That's why I didn't do it in a first place: there's simply no way to copy and paste `Summary` approach into `Histogram`.

Reply:
Ah, yes. That's my bad. In that case I would remove the functionality in its entirety. As per [swift-metrics](https://github.com/apple/swift-metrics/blob/main/Sources/CoreMetrics/Metrics.swift#L555-L559):

```
    /// Set the preferred display unit for this TimerHandler.
    ///
    /// - parameters:
    ///     - unit: A hint to the backend responsible for presenting the data of the preferred display unit. This is not guaranteed to be supported by all backends.
```